kube-client/src/config/file_config.rs

```diff
@@ -525,6 +541,60 @@ impl AuthInfo {
     }
 }
 
+/// Cluster stores information to connect Kubernetes cluster used with auth plugins
+/// that have `provideClusterInfo`` enabled.
+/// This is a copy of [`kube::config::Cluster`] with certificate_authority passed as bytes without the path.
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct ExecAuthCluster {
```

Comment:
Does this struct have an upstream variant somewhere or is it more here to try to replicate how client-go is passing cluster data to exec plugins?

Reply:
It does have an upstream variant:
https://github.com/kubernetes/client-go/blob/master/pkg/apis/clientauthentication/types.go#L80

Reply:
but that's the main cluster struct though, right? the one that typically represents a `~/.kube/config` entry under `.clusters[]`

Reply:
oh, wait it's not.
- main cluster struct :: https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/api/types.go#L66-L104
- client exec cluster :: https://github.com/kubernetes/client-go/blob/master/pkg/apis/clientauthentication/types.go#L73-L129